### PR TITLE
feat: video player & audio player

### PR DESCRIPTION
**Pull request by account0123**
### Video player
In message, video opens as preview. Full-screen control activates full mode (it pauses preview and opens video on modal).
Modal contains same header as image viewer modal, with similar actions.
I haven't implemented video player in MessageEmbed yet.
### Audio player
It was hard to find an audio player component. I tried installing a [voice message player](https://www.npmjs.com/package/@carchaze/react-native-voice-message-player) but installation failed. Then, I built my own component using [TrackPlayer](https://www.npmjs.com/package/react-native-track-player), a service useful for music queues, and I implemented a method to ensure that only one audio is played at a time.
Some known bugs:
- No communication from remote controls (on notifications) to audio attachment. But background track can be stopped easily.
- On player setup, the app could sometimes quit. This behavior happens once without explanation.

## Please make sure to check the following tasks before opening and submitting a PR

* [X] I understand and have followed the [contribution guide](https://github.com/revoltchat/revolt/discussions/282)
* [X] I have tested my changes locally and they are working as intended
* [X] These changes do not have any notable side effects on other Revolt projects

### CODE REVIEW

**Comment by Rexogamer** [comment]
out of curiosity, what failed when trying to install the first library?

**Comment by account0123** [comment]
```shell
$ yarn add @carchaze/react-native-voice-message-player
<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
 1: 0x55916db37f30 node::Abort() [node]
 2: 0x55916da30f4b  [node]
 3: 0x55916dd538a0 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [node]

```

**Comment by Rexogamer** [comment]
ahhh. there seems to be an issues where the latest releases of React Native - and, by extension, packages that depend on them - can't be installed due to a Yarn bug; your alternate solution should be fine (although it'd be preferable to fix the known issues first), and the seeming solution to this issue is to switch to Yarn Berry (I'll wait for v4 to release) so no worries

**Comment by Rexogamer** [comment]
Conflicting

**Comment by account0123** [comment]
Hi! As you said before I tried to install (in new branch) `@carchaze/react-native-voice-message-player`. Installation was successful, but in execution time it has a very [weird bug](https://github.com/zmxv/react-native-sound/issues/592) related to `react-native-sound` dependency. I followed steps but nothing worked so I give up trying to install that library.
On other hand I am thinking of creating my own native sound module instead of using `react-native-track-player` because it's not necessary to use background audio service that this dependency requires. So in the new branch I am going to experiment with audio native modules or creating my own module and create a PR when everything works fine.
TL;DR: I will fix conflicts keeping only video components. Audio components will be in another PR.

**Comment by Rexogamer** [comment]
I will also mention that you might be able to follow the steps [here](https://github.com/zmxv/react-native-sound/pull/686/files) if you haven't already - however, if you have to no avail, no worries. Take as much time as you need ^^

**Comment by account0123** [comment]
hehe that task doesnt exist in the gradle project, then I manually deleted ~/.gradle/cache

File: src/components/common/messaging/index.ts [hunk 1]
```diff
@@ -4,3 +4,5 @@ export {MessageEmbed} from './MessageEmbed';
 export {MessageReactions} from './MessageReactions';
 export {PlatformModerationMessage} from './PlatformModerationMessage';
 export {ReplyMessage} from './ReplyMessage';
+export {VideoEmbed} from './VideoEmbed';
+export {AudioPlayer} from './AudioPlayer';
```
Comment (Rexogamer) [inline src/components/common/messaging/index.ts:8]: With the splitting of this PR, this needs to be removed

File: src/lib/utils.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import {ToastAndroid} from 'react-native';
+import TrackPlayer, {Event} from 'react-native-track-player';
```
Comment (Rexogamer) [inline src/lib/utils.ts:2]: ditto

**Review by Rexogamer — CHANGES_REQUESTED** [review]
conflicts + a couple remnants; will review the main bit once this is done

**Comment by Rexogamer** [comment]
closing this as stale. if you want to keep working on this, please feel free to reopen it (or nudge me if you can't). thanks for contributing :3
